### PR TITLE
New Class for LoadSpec, with additional properties

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: core/refresh/LoadSpec.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    owner;
+
+    get isLoadSpec()    {return true}
+
+    get isLatest()      {
+        return this === this.owner._lastRequested;
+    }
+
+    get isObsolete()    {
+        return this.owner._lastSucceeded.loadNumber > this.loadNumber;
```
Comment (haynesjm42) [inline core/refresh/LoadSpec.js:48]: `_lastSucceeded` could be `null` on the owner here

File: core/refresh/LoadSpec.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    /**
+     *  @private
+     *  Not for application use.  Used by LoadSupport.
+     */
+    constructor({loadNumber, isRefresh, isAutoRefresh, owner}) {
+        this.loadNumber = loadNumber;
```
Comment (haynesjm42) [inline core/refresh/LoadSpec.js:56]: Where is `loadNumber` ever being passed in by calling code? Should we just set it in the ctor here based on `owner._lastRequested`?

File: core/refresh/LoadSpec.js (hunk 1 above)
Comment (lbwexler) [reply core/refresh/LoadSpec.js:48]: great catch

File: core/refresh/LoadSpec.js (hunk 2 above)
Comment (lbwexler) [reply core/refresh/LoadSpec.js]: fixed

**Review by amcclain — APPROVED** [review]
Looks good to me, thanks!  Please verify that the changes + one fix I made look correct and complete.

Any opportunities to check for stale loads in Hoist or Toolbox? Would be great to have an example to point to.  (Thinking Toolbox could support a dedicated tab on LoadSupport, given that it is both really important/powerful but also not all that obvious - but we certainly should not block on that.)

**Pull request merged**